delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+                    health = data.get('health', '')
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    total_size = self.parse_string(data.get('total-size'))
+                    data_map = {
+                        'name': data.get('name', ''),
+                        'storage_id': storage_id,
+                        'native_storage_pool_id': data.get('serial-number',
+                                                           ''),
+                        'description': '',
+                        'status': status,
+                        'storage_type': constants.StorageType.BLOCK,
+                        'total_capacity': int(total_size),
+                        'subscribed_capacity': int(blocks),
+                        'used_capacity': volume_size,
+                        'free_capacity': int(total_size - volume_size)
+                    }
+                pool_arr.append(data_map)
+            return pool_arr
+        except Exception as e:
+            err_msg = "Failed to get storage pool: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def parse_string(self, value):
+        capacity = 0
+        if value:
+            if value.isdigit():
+                capacity = float(value)
+            else:
+                if value == '0B':
+                    capacity = 0
+                else:
+                    unit = value[-2:]
+                    capacity = float(value[:-2]) * int(
+                        Tools.change_capacity_to_bytes(unit))
+        return capacity
+
+    def handle_detail(self, deltail_info, element):
+        detail_arr = []
+        detail_data = deltail_info.split('\n')
+        detail = detail_data[1:len(detail_data) - 1]
+        detail_xml = ''.join(detail)
+        xml_element = ET.fromstring(detail_xml)
+        for ch in xml_element.iter("OBJECT"):
+            property_name = ch.get('basetype')
+            if element == property_name:
+                msg = {}
+                for child in ch.iter("PROPERTY"):
+                    msg[child.get('name')] = child.text
+                detail_arr.append(msg)
+        return detail_arr
+
+    def list_storage_error(self, query_para):
+        alert_list = []
+        try:
+            error_infos = self.ssh_pool.do_exec('show events error')
+            err_arr = self.handle_detail(error_infos, 'events')
+            if err_arr:
+                for alert_map in err_arr:
+                    time_stamp = alert_map.get('time-stamp-numeric')
+                    occur_time = int(time_stamp) * self.SECONDS_TO_MS
+                    if not alert_util.is_alert_in_time_range(query_para,
+                                                             occur_time):
+                        continue
+                    alert_name = alert_map.get('message', '')
```

Comment:
Default values of below 3 `get` functions are unnecessary.

Reply:
The default value is cancelled